src/renderers/shaders/ShaderChunk/bsdfs.glsl.js

```diff
@@ -66,8 +66,37 @@ float D_GGX( const in float alpha, const in float dotNH ) {
 
 }
 
+#ifdef USE_ANISOTROPY
```

Comment:
I'd suggest you avoid placing this here. Copy the `transmission` pattern and create:
```
anisotropy_pars_fragment
anisotropy_fragment
```

Reply:
Okay, I'll look into that tomorrow.

Reply:
@WestLangley Okay, now that I've got a version of this functional I did look into this a bit. Respectfully, can you give me a bit of guidance on what you consider good patterns to be here and why? It might help me a bit with my reasoning, especially since I'm proposing some refactors here. 

What I did so far was to follow more the pattern of `sheen` and `specular`, since those seemed to be more similar to this extension in terms of the small amount of code added. Also, I personally find it rather difficult to follow the code flow when functions are scattered across files, hence why I moved the BRDFs that are only called by `lights_physical_pars_fragment` into that file (but of course that file is also getting large now...). Likewise, I find it much clearer to have the uniforms defined together as in `meshphysical` than separately in e.g. `specularmap_pars_fragment` or `iridescence_pars_fragment`. However, I could easily be missing a goal state that you may be working towards.

Reply:
@elalish Thanks for this. I have been following along, and your changes seem acceptable to me so far.

But quite frankly, the shader code is fragile and difficult to both comprehend and maintain. I also understand there is a desire to switch to `NodeMaterial`. Significant leadership is needed in this area of the project.

Reply:
@WestLangley We're still figuring out `NodeMaterial` and it's also dependent on WebGPU/WGSL. As soon as things settle down a bit on that side we can start moving to it. But right now I think it's best to continue dealing with ShaderChunks.

Reply:
Also... `NodeMaterial` doesn't use the fragile ShaderChunks code, but neither does it use the familiar GLSL.

Reply:
> it's also dependent on WebGPU/WGSL

It compiles to GLSL just in the same way as to WGSL.

> but neither does it use the familiar GLSL.

Yes, it is a separate language in some sense -- but it is much easier to use than GLSL or WGSL's chunks.

Reply:
>NodeMaterial doesn't use the fragile ShaderChunks code

@mrdoob `NodeMaterial` is based on that code, right? Every time we refactor the shader code -- or add to it -- it impacts @sunag.

Reply:
> Yes, it is a separate language in some sense -- but it is much easier to use than GLSL or WGSL's chunks.

But it's still in development.

I do not think it makes sense to ask contributors to learn that language if they want to contribute new GLTF properties until the language is more finalized.

> @mrdoob NodeMaterial is based on that code, right? Every time we refactor the shader code -- or add to it -- it impacts @sunag.

Yes, we'll have to port any code added to ShaderChunks to `NodeMaterial`.
I think that's one of the things @sunag is working on.

Reply:
I don't think this kind of change is doable for `WebGLRenderer`. It will break too many things.

Reply:
But will also allow so much new things to be done in much cleaner and simpler ways... Maybe we can create a new version of WebGLRenderer and then after some time replace the current one with it...

Reply:
Maybe a crazy idea but...

How about creating a `WebGPUFallbackRenderer` (using WebGL) and it is automatically used when the developer is using `WebGPURenderer` but the system doesn't support `WebGPU`.

That fallback renderer would live inside `WebGPURenderer` until it's no longer needed (WebGPU supported everywhere).

That fallback renderer would only support `NodeMaterial`.

Reply:
I don't think it would be a good idea to consider WebGL just as a fallback when WebGPU doesn't work... I think we should just create a new version of WebGLRenderer (either called the same or WebGLNodesRenderer) and then at some time deprecated the current WebGLRenderer and replace it with the new one.

Reply:
I think you said the same thing I did but with different words 🤓

Reply:
Just had a quick chat with @sunag about this and he has better names for this solution.
I'll let him explain when he finds the time.

Reply:
I mean that we shouldn't think that WebGL is just a fallback for WebGPU -- we should continue to support them both.

Reply:
We can use this transition in our advantage:

* `WebGLRenderer` would be something like Three.js v1.0
  * Deprecated eventually
* `WebGPURenderer` would be something like Three.js v2.0
  * Not fully backwards compatible
  * Node based
  * WebGL fallback

I doubt we'll be able to support `ShaderMaterial` in `WebGPURenderer`...

Neither I know yet how `WebGPURenderer` imports should be.

Maybe `import { WebGPURenderer } from 'three-webgpu'`?

Reply:
> I doubt we'll be able to support ShaderMaterial in WebGPURenderer...

In a new WebGLRenderer we wouldn't need ShaderMaterial, because everything would be possible to do with the Nodes system.
We should have WebGPURenderer and not-backwards-compatible WebGLRenderer, I agree with that -- but they should be two different and equal renderers, not one being a fallback for other.

Reply:
So what @sunag was suggesting was to have something like:

```
                 - WebGPUBackend
WebGPURenderer -| 
                 - WebGLBackend
```

We could rename `WebGPURenderer` to just `Renderer` but I think it will be easier for devs to understand that `WebGLRenderer` (`three`) is the current ecosystem and `WebGPURenderer` (`three-webgpu`) is the next ecosystem.

And that `WebGPURenderer` will have an internal WebGL backend/fallback to ensure maximum support.

Similar to how `WebGLRenderer` internally uses WebGL 2.0 by default and WebGL 1.0 as fallback.

The developer shouldn't have to know about these details.

Reply:
> Similar to how WebGLRenderer internally uses WebGL 2.0 by default and WebGL 1.0 as fallback.

This is possible because WebGL 2 is backwards-compatible with WebGL 1, so 'WebGL2Renderer" is compatible with "WebGL1Renderer".
It's better to let the developer to choose between WebGL and WebGPU renderers rather than hiding the implementation from them.